src/array/ops/cast.rs

```diff
@@ -701,12 +705,120 @@ impl EmbeddingArray {
 
 impl ImageArray {
     pub fn cast(&self, dtype: &DataType) -> DaftResult<Series> {
-        self.physical.cast(dtype)
+        match dtype {
+            #[cfg(feature = "python")]
+            DataType::Python => {
+                Python::with_gil(|py| {
+                    let mut ndarrays = Vec::with_capacity(self.len());
+                    let mut bitmap = arrow2::bitmap::MutableBitmap::with_capacity(self.len());
+                    for i in 0..self.len() {
+                        let img_buf = self.as_image_obj(i);
+                        let ndarray = img_buf
+                            .map(|buf| {
+                                let slice = buf.as_u8_slice();
+                                let shape = (
+                                    buf.height() as usize,
+                                    buf.width() as usize,
+                                    buf.mode().num_channels() as usize,
+                                );
+                                // NOTE(Clark): This will create a copy in the Python heap, using NumPy for allocation.
+                                PyArray::from_slice(py, slice).reshape(shape.into_dimension())
```

Comment:
I just looked into it and it seems that going through the pyo3 / numpy path is gonna be a bunch of work and that pyarrow already implements it: https://github.com/apache/arrow/blob/f44f7685638e4afb97f8d6d536d8538b5292b47d/python/pyarrow/src/arrow/python/arrow_to_pandas.cc#L316

Reply:
A suggestion here is to slice the parent arrow2 data array and then go through ffi to pyarrow and call `to_numpy` that way we get the zero copy numpy array.

Reply:
I had assumed that we didn't do that for the ingress side (and that we don't want to do that for the egress side) so we could have more control over the conversion, e.g. whether we want to create a borrowed immutable view vs. an owned mutable copy, and to add some later perf optimizations. But if that's not a priority, leveraging pyarrow would definitely be simpler.

I'll push up a commit with that implementation.